src/main/kotlin/blackjack/CardFactory.kt

```diff
@@ -0,0 +1,16 @@
+package blackjack
+
+object CardFactory {
+    fun from(
+        cardNumber: CardNumber,
+        suit: Suit,
+    ): Card = Card(cardNumber, suit)
+
+    fun all(): List<Card> {
+        return Suit.entries.flatMap { suit ->
+            CardNumberFactory.all().map { cardNumber ->
+                Card(cardNumber, suit)
+            }
+        }
+    }
```

Comment:
행위가 factory 객체로 옮겨졌지만 엔티티 그래프 탐색이 깊다는 부분에서는 동일합니다. 코틀린은 함수형 프로그래밍이 수월하기 때문에 좀 더 책임 위임을 하면 어떨까 라는 의견은 팩토리객체로 새로 만들어서 옮기라는 피드백보다는 행위 자체를 더 적절한 객체에 위임시켜서 엔티티 그래프 탐색의 깊이를 낮추라는 의도였습니다. 

```suggestion
        return Suit.flatMap { suit ->
            CardNumberFactory.mapValue { cardNumber ->
                Card(cardNumber, suit)
            }
        }
    }
```